go/runtime/flow_consumer.go

```diff
@@ -297,7 +298,12 @@ func shardApply(ctx context.Context, svc *consumer.Service,
 		if nextSpec != nil && logSpec == nil {
 			// Grab labeled catalog task and its journal rules.
 			var name = nextSpec.LabelSet.ValueOf(labels.TaskName)
-			var _, commons, _, err = catalog.GetTask(name)
+			var revisionStr = nextSpec.LabelSet.ValueOf(labels.CatalogRevision)
+			var catalogRevision, err = strconv.ParseInt(revisionStr, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("parsing catalog revision: %w", err)
+			}
+			_, commons, _, err := catalog.GetTaskWithRevision(ctx, name, catalogRevision)
```

Comment:
Oh! It's not even about the reactor starting the shard and not seeing the catalog task -- it's about `flowctl apply` driving the shard Apply RPC before the reactor has read-through the catalog.

This is gross, but I actually am thinking that simply adding a `time.Sleep` in `flowctl apply` to account for this is the right move right now. Wiring up this revision as a shard label works, but is feeling weird to me -- that we haven't gotten the right API. And we're likely to change the means by which the catalog is applied to Etcd anyway while building out the control plane. As an example (I don't think we're doing it this way, but:) if the reactor actually applied catalog tasks to Etcd, then it could trivially read-through the applied catalog KeySpace revision before returning from its RPC, gating a future Shard apply RPC, and this race wouldn't be a problem.